app/controllers/statements_controller.rb

```diff
@@ -50,15 +45,8 @@ def init_account
   #
   # Override CanCan's find -- it won't properly search by 'recent'
   def init_statement
-    @facility = Facility.find_by_url_name!(params[:facility_id])
-    @statements = @account.statements.where(facility_id: @facility.id)
-
-    @statement = if params[:id] =~ /\w+/i
-                   @statements.blank? ? Statement.find_by_facility_id(@facility.id) : @statements.first
-                 else
-                   @account.statements.find(params[:id])
-                 end
-    @statement = Statement.new if @statement.nil?
+    @statement = Statement.find_by(id: params[:id], account_id: @account.id)
```

Comment:
`find_by!` might be a good idea here so it goes straight to a 404 rather than 500ing on the next line if it's not found.